lib/OpenQA/Schema/Result/Assets.pm

```diff
@@ -172,15 +174,16 @@ sub limit_assets {
             $seen_asset{$a->asset->id} = $g->id;
             my $size = $a->asset->ensure_size;
             if ($size > 0 && $sizelimit > 0) {
-                $keep{$a->asset_id} = 1;
+                $keep{$a->asset_id} = sprintf "%s: %s/%s", $g->name, $a->asset->type, $a->asset->name;
             }
             else {
-                $toremove{$a->asset_id} = 1;
+                $toremove{$a->asset_id} = sprintf "%s/%s", $a->asset->type, $a->asset->name;
             }
             $sizelimit -= $size;
         }
     }
     for my $id (keys %keep) {
+        OpenQA::Utils::log_debug("KEEP $toremove{$id} $keep{$id}") if $debug_keep;
```

Comment:
This would report on uninitialized variable, wouldn't it? There is always either `$toremove{$id}` or `$keep{$id}` but not both, or am I missing something?